lib/crypto/src/crypto.erl

```diff
@@ -47,533 +56,572 @@
+    pk_sign_verify_opts/0,
+    rsa_digest_type/0,
+    sha1/0,
+    sha2/0
+]).
+
+-export_type([
+    engine_ref/0,
+    key_id/0,
+    password/0
+]).
 
 %%% Opaque types must be exported :(
 -export_type([
-              stream_state/0,
-              hmac_state/0,
-              hash_state/0,
-              crypto_state/0,
-              mac_state/0
-             ]).
+    stream_state/0,
+    hmac_state/0,
+    hash_state/0,
+    crypto_state/0,
+    mac_state/0
+]).
 
 %% Private. For tests.
--export([packed_openssl_version/4, engine_methods_convert_to_bitmask/2,
-	 get_test_engine/0]).
+-export([
+    packed_openssl_version/4,
+    engine_methods_convert_to_bitmask/2,
+    get_test_engine/0
+]).
+
 -export([rand_plugin_aes_jump_2pow20/1]).
 
 -deprecated({rand_uniform, 2, "use rand:uniform/1 instead"}).
 
 %% This should correspond to the similar macro in crypto.c
--define(MAX_BYTES_TO_NIF, 20000). %%  Current value is: erlang:system_info(context_reductions) * 10
+
+%%  Current value is: erlang:system_info(context_reductions) * 10
+-define(MAX_BYTES_TO_NIF, 20000).
 
 %% Used by strong_rand_float/0
--define(HALF_DBL_EPSILON, 1.1102230246251565e-16). % math:pow(2, -53)
 
+% math:pow(2, -53)
+-define(HALF_DBL_EPSILON, 1.1102230246251565e-16).
 
 %%% ===== BEGIN NEW TYPING ====
 
 %%% Basic
--type key_integer() :: integer() | binary(). % Always binary() when used as return value
-
-%%% Keys
--type rsa_public() :: [key_integer()] . % [E, N]
--type rsa_private() :: [key_integer()] . % [E, N, D] | [E, N, D, P1, P2, E1, E2, C]
--type rsa_params() :: {ModulusSizeInBits::integer(), PublicExponent::key_integer()} .
-
--type dss_public() :: [key_integer()] . % [P, Q, G, Y]
--type dss_private() :: [key_integer()] . % [P, Q, G, X]
-
--type ecdsa_public()  :: key_integer() .
--type ecdsa_private() :: key_integer() .
--type ecdsa_params()  :: ec_named_curve() | ec_explicit_curve() .
-
--type eddsa_public()  :: key_integer() .
--type eddsa_private() :: key_integer() .
--type eddsa_params()  :: edwards_curve_ed() .
-
--type srp_public() :: key_integer() .
--type srp_private() :: key_integer() .
--type srp_gen_params()  :: {user,srp_user_gen_params()}  | {host,srp_host_gen_params()}.
--type srp_comp_params() :: {user,srp_user_comp_params()} | {host,srp_host_comp_params()}.
--type srp_user_gen_params() :: list(binary() | atom() | list()) .
--type srp_host_gen_params() :: list(binary() | atom() | list()) .
--type srp_user_comp_params() :: list(binary() | atom()) .
--type srp_host_comp_params() :: list(binary() | atom()) .
 
--type dh_public() :: key_integer() .
--type dh_private() :: key_integer() .
--type dh_params() :: [key_integer()] . % [P, G] | [P, G, PrivateKeyBitLength]
+% Always binary() when used as return value
+-type key_integer() :: integer() | binary().
 
--type ecdh_public()  :: key_integer() .
--type ecdh_private() :: key_integer() .
--type ecdh_params()  :: ec_named_curve() | edwards_curve_dh() | ec_explicit_curve() .
+%%% Keys
```

Comment:
Does `%%%` have special meaning?

Reply:
Yes, three % makes Emacs put the comment at the beginning of the line.